components/UI/Datalist.js

```diff
@@ -0,0 +1,114 @@
+        setValue(e.target.value);
+        onSelect(null);
+        await onChange(e.target.value);
+        if (e.target.value !== '') setIsOpen(true);
+        else {
+            // if the input is empty we close the datalist
+            setIsOpen(false);
+        }
+        setIsLoading(false);
+    };
+
+    const handleKeyDown = (e) => {
+        // Handle keyboard navigation
+        if (e.key === 'ArrowDown') {
+            e.preventDefault();
+            if (!isOpen) {
+                setIsOpen(true);
+            } else {
+                const newIndex = Math.min(selectedIndex + 1, options.length - 1);
+                setSelectedIndex(newIndex);
+            }
+        } else if (e.key === 'ArrowUp') {
+            e.preventDefault();
+            if (isOpen) {
+                const newIndex = Math.max(selectedIndex - 1, -1);
+                setSelectedIndex(newIndex);
+            }
+        } else if (e.key === 'Enter' && isOpen && selectedIndex !== -1) {
+            e.preventDefault();
+            const selectedOption = options[selectedIndex];
+            handleSelect(selectedOption);
+        }
+    };
+
+    const handleBlur = () => {
+        setTimeout(() => {
+            setIsOpen(false);
+            setSelectedIndex(-1);
+        }, 100); // Delay closing the datalist to allow clicking on options
+    };
+
+    const handleSelect = (selectedOption) => {
+        setValue(stringifySelection(selectedOption));
+        onSelect(selectedOption);
+        setSelectedIndex(-1);
+        setIsOpen(false);
+        inputRef.current.focus();
+    };
+
+    const stringifySelection = (selectedOption) => {
+        return keysToDisplay
+            .map((key) => selectedOption[key])
+            .filter((value) => value !== undefined)
+            .join(' (') + ')'; // wrap in brackets
+    };
+
+    return (
+        <div>
+            <input
+                type="text"
+                value={value}
+                onChange={handleChange}
+                onKeyDown={handleKeyDown}
+                onBlur={handleBlur}
+                ref={inputRef}
+                disabled={isLoading}
+            />
+            { options.length > 0 && isOpen && (
+                <ServiceTable>
+                    { options.map((option, index) => (
+                        <ServiceTable.Row
+                            key={index}
+                            selected={selectedIndex === index}
+                            onClick={() => handleSelect(option)}>
+                            { keysToDisplay.map(key => {
+                                return <ServiceTable.Cell
+                                    key={key}>
+                                    { option[key] }
+                                </ServiceTable.Cell>;
+                            }) }
+
+                        </ServiceTable.Row>
+                    )) }
+                </ServiceTable>
+            ) }
+        </div>
+    );
+}
+
+export default Datalist;
```

Comment:
When there's not a lot of top-level stuff happening in a file -- like in this one -- you can do this all in one line:

```js
export default function Datalist({...}) {
}
```

This is also what the React documentation recommends: https://react.dev/learn/your-first-component#defining-a-component

Please adjust the other files accordingly. (`DefaultModal`, `UserListEntry`)

Going to see if I can get ESLint to remind us of this in the future.